kornia/geometry/transform/flips.py

```diff
@@ -123,8 +123,8 @@ def hflip(input: torch.Tensor) -> torch.Tensor:
         torch.Tensor: The horizontally flipped image tensor
 
     """
-
-    return torch.flip(input, [-1])
+    w = input.shape[-1]
+    return input[..., torch.arange(w - 1, -1, -1)]
```

Comment:
have you tested in GPU ? this I guess will crash, 
```python
w = input.shape[-1]
index = torch.arange(w - 1, -1, -1, device=input.device, dtype=input.dtype)
return input[..., index]
```

Reply:
Yes. It makes more sense.

@FGeri Can you do another PR for a quick fix? I do not have a GPU machine by hand.